src/main/scala/com/amazon/deequ/analyzers/PatternMatch.scala

```diff
@@ -44,9 +45,9 @@ case class PatternMatch(column: String, pattern: Regex, where: Option[String] =
   }
 
   override def aggregationFunctions(): Seq[Column] = {
-
-    val expression = when(regexp_extract(col(column), pattern.toString(), 0) =!= lit(""), 1)
-      .otherwise(0)
+    val expression = when(
+        regexp_extract(col(ColumnName.sanitize(column)), pattern.toString(), 0) =!= lit(""), 1
+    ).otherwise(0)
```

Comment:
please put ColumnName.sanitize(column) into a temporary variable as this expression is becoming to complex otherwise

Reply:
I think we should break this up into a few more statements, still looks to complicated :)